xarray/core/merge.py

```diff
@@ -365,6 +365,24 @@ def merge_data_and_coords(data, coords, compat='broadcast_equals',
     return merge_core(objs, compat, join, explicit_coords=explicit_coords)
 
 
+def assert_valid_explicit_coords(variables, explicit_coords):
+    """Validate explicit coordinate names/dims.
+
+    Raise a MergeError if an explicit coord shares a name with a dimension
+    but is comprised of arbitrary dimensions.
+    """
+    var_dims = []
+    for name, var in variables.items():
+        if name not in explicit_coords:
+            var_dims.extend(var.dims)
+    for coord_name in explicit_coords:
+        if coord_name in var_dims and not all(
+                [d in var_dims for d in variables[coord_name].dims]):
```

Comment:
Rather than `not all(...)`, I think this condition should be just `variables[coord_name].dims != (coord_name,)`

Reply:
yes, this works.